packages/flutter/lib/src/services/binding.dart

```diff
@@ -43,6 +43,7 @@ mixin ServicesBinding on BindingBase, SchedulerBinding {
     SystemChannels.system.setMessageHandler((dynamic message) => handleSystemMessage(message as Object));
     SystemChannels.lifecycle.setMessageHandler(_handleLifecycleMessage);
     SystemChannels.platform.setMethodCallHandler(_handlePlatformMessage);
+    _registerBindingWithEngine();
```

Comment:
We may want to leverage this as a generic "the `ServicesBinding` is fully initialized" message in the future. Should we make this be the last step in `initInstances`?

Reply:
On second thought, it seems weird to require an app to send a `System.registerBinding` if it wants to listen to application lifecycle events. For example, a `dart:ui` app may want to listen to app lifecycle events without using `ServicesBinding`. Perhaps we should use a different name like `System.enableApplicationLifecycle`?

Reply:
I don't think it's that weird. Having something that happens after the binding is initialized seems like something you might want to use for other things besides enabling application lifecycle events too, so it doesn't seem too odd to have a "we're done initializing the binding" message to start things off. It does increase the boilerplate for such an app, however, which is too bad, but I think anyone building an app at that level is expecting some level of boilerplate.

Reply:
To clarify, I'm fine with the boilerplate. I disliked the previous message name `System.registerBinding` for a raw `dart:ui` app.

Reply:
How about `initializationComplete`? Is that really true at the point the message is sent?

Reply:
I'd be fine with `System.initializationComplete` too, but what would be the best way to document its implications? (Namely that the app now needs to respond to `System.requestAppExit`)

Reply:
Yes, for now that is all it implies: that it's ready to respond to  `System.requestAppExit` requests, but to the embedding, it also represents the point at which the framework's bindings are registered, so it can feel free to communicate with the bindings. I also wonder if there are other parts of the engine that assume the binding will respond to a method call which would benefit from waiting until the binding is ready?